js/fetch.ts

```diff
@@ -97,59 +99,127 @@ export const Headers = DomIterableMixin<string, string, typeof HeadersBase>(
   headerMap
 );
 
-class FetchResponse implements domTypes.Response {
+class Body implements domTypes.Body, domTypes.ReadableStream, io.Reader {
+  bodyUsed = false;
+  readonly body: domTypes.ReadableStream | null = null; // TODO
+  private _buffering = false;
+  private _bodyWaiter: Resolvable<ArrayBuffer>;
+  private _chunks: Uint8Array[] = [];
+  private _data: Uint8Array | null = null;
+  readonly locked: boolean = false; // TODO
+
+  constructor(readonly rid: number, readonly contentType: string) {
+    this._bodyWaiter = createResolvable();
+  }
+
+  private async _bufferBody(): Promise<void> {
+    if (!this._buffering) {
+      this._buffering = true;
+      try {
+        while (true) {
+          const b = new Uint8Array(64 * 1024);
+          const r = await read(this.rid, b);
+          if (r.nread > 0) {
+            this._chunks.push(b.subarray(0, r.nread));
+          }
+          if (r.eof) {
+            break;
+          }
+        }
+        this._data = concatUint8Array(this._chunks);
+        this._chunks = [];
+        this._bodyWaiter.resolve(this._data!.buffer as ArrayBuffer);
```

Comment:
I don't think you need ! here, nor that type assertion.

Reply:
Removed the ! - good catch

If I remove the cast, I get:
```
[!] (rpt2 plugin) Error: ../../js/fetch.ts:131:34 - error TS2345: Argument of type 'ArrayBuffer | SharedArrayBuffer' is not assignable to parameter of type 'ArrayBuffer | PromiseLike<ArrayBuffer> | undefined'.
  Type 'SharedArrayBuffer' is not assignable to type 'ArrayBuffer | PromiseLike<ArrayBuffer> | undefined'.
    Type 'SharedArrayBuffer' is not assignable to type 'PromiseLike<ArrayBuffer>'.
      Property 'then' is missing in type 'SharedArrayBuffer'.

131         this._bodyWaiter.resolve(this._data.buffer);
```